locustempus/main/views.py

```diff
@@ -230,91 +230,109 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
         }
         send_template_email(
             'Locus Tempus Course Connected',
             'main/notify_lti_course_connect.txt',
             data, settings.SERVER_EMAIL)
 
-    def thank_faculty(self, course):
+    def thank_faculty(self, course: Course) -> None:
+        user = self.request.user
         send_template_email(
             'Locus Tempus Course Connected',
             'main/lti_course_connect.txt',
-            {'course': course}, self.request.user.email)
+            {'course': course},
+            user.email if user.email else user.username + '@columbia.edu')
 
-    def groups_from_context(self, course_context):
+    def groups_from_context(self, course_context) -> Tuple[Group, Group]:
         group, created = Group.objects.get_or_create(name=course_context)
         faculty_group, created = Group.objects.get_or_create(
             name='{}_faculty'.format(course_context))
         return (group, faculty_group)
 
-    def groups_from_sis_course_id(self, attrs):
+    def groups_from_sis_course_id(self, attrs) -> Tuple[Group, Group]:
+        user = self.request.user
         st_affil = WindTemplate.to_string(attrs)
         group, created = Group.objects.get_or_create(name=st_affil)
-        self.request.user.groups.add(group)
+        user.groups.add(group)
 
         attrs['member'] = 'fc'
         fc_affil = WindTemplate.to_string(attrs)
         faculty_group, created = Group.objects.get_or_create(name=fc_affil)
-        self.request.user.groups.add(faculty_group)
+        user.groups.add(faculty_group)
         return (group, faculty_group)
 
-    def get_year_and_term_from_sis_course_id(self, sis_course_id):
-        m = re.match(
+    def add_yt_to_course(self, sis_course_id: str, course: Course) -> None:
+        """
+        Sets the year and term attributes on a course if
+        they can be deteremend from a sis_course_id
+        """
+
+        # CanvasTemplate matches a CU course string
+        cu_course = CanvasTemplate.to_dict(sis_course_id)
+        # TC courses use a different format
+        tc_course = re.match(
             (r'(?P<year>\d{4})(?P<term>\d{2})'), sis_course_id)
-        if m:
-            return m.groupdict()
+
+        if cu_course:
+            course.info.term = cu_course['term']
+            course.info.year = cu_course['year']
+            course.info.save()
+        elif tc_course:
+            course.info.term = tc_course['term']
+            course.info.year = tc_course['year']
+            course.info.save()
 
     def post(self, *args, **kwargs):
+        user = self.request.user
         course_context = self.request.POST.get('lms_course')
         title = self.request.POST.get('lms_course_title')
-
-        sis_course_id = self.request.POST.get('sis_course_id', '')
-        d = CanvasTemplate.to_dict(sis_course_id)
-
-        if d:
-            (group, faculty_group) = self.groups_from_sis_course_id(d)
+        sis_course_id = self.request.POST['sis_course_id'] \
```

Comment:
Why not use `self.request.POST.get('sis_course_id', None)` here?

Reply:
I needed this to coerce the type from a string `'None'` to `None`. TBH I was surprised that Django didn't do this for me. Maybe I'm missing something.
I do think I could make the logic more clear by swapping the clauses, let me try that.